### PR TITLE
Eslint: Fixed 'angular/controller-as' violations in Blueprint Editor and Canvas

### PR DESCRIPTION
**Pull request by dtaylor113**
No visual differences in GUI, just improving the source code :smile: 

This PR contains #186, please merge #186 first!

Also fixed 'eslint consistent-this' in Blueprint Order List Service.

@AllenBW , @chriskacerguis , @serenamarie125

### CODE REVIEW

File: client/app/states/designer/blueprints/editor/editor.html [hunk 1]
```diff
@@ -1,15 +1,14 @@
-<span ng-controller="BlueprintEditorController" ng-class="{'wait-cursor': loading}">
+<span ng-controller="BlueprintEditorController as evm" ng-class="{'wait-cursor': loading}">
```
Comment (AllenBW) [inline client/app/states/designer/blueprints/editor/editor.html]: ooh ooh editor view model and canvas view model!  I get it <3

one of these days we'll have to componentize these puppies... one of these days...

File: client/app/components/flowchart/flowchart_directive.js [hunk 2]
```diff
@@ -82,7 +82,7 @@ angular.module('flowChart', ['dragging'] )
 // it is painful to unit test a directive without instantiating the DOM
 // (which is possible, just not ideal).
 //
-.controller('FlowChartController', ['$scope', 'dragging', '$element', function FlowChartController($scope, dragging, $element) {
+.controller('FlowChartController', ['$scope', 'dragging', '$element', '$document', function FlowChartController($scope, dragging, $element, $document) {
```
Comment (AllenBW) [inline client/app/components/flowchart/flowchart_directive.js]: Looks like `$scope` could be entirely removed from this controller (and dependency list)

Comment (dtaylor113) [inline client/app/components/flowchart/flowchart_directive.js]: Umm..$scope in FlowChartController is still being used.  I haven't yet attempted to clean up flowchart_directive.js.  It still has a /* eslint-disable */ at the top.

**Comment by chriskacerguis** [comment]
@dtaylor113 If you could please do not do a "please merge X before this ticket" I would appreciate that a great deal.  It becomes too much of a pain to keep track of that.

File: client/app/states/designer/blueprints/editor/editor.html (hunk 1 above)
Comment (dtaylor113) [inline client/app/states/designer/blueprints/editor/editor.html]: I'd welcome any feedback in that regard.  The "Editor" controls the page level buttons/actions, while the "Canvas" controls the node CRUD; they are very much intertwined.  My only thought would be to make them two directives (?)

**Comment by dtaylor113** [comment]
@chriskacerguis is that a GitHub thing?!  I have never filed a 'merge X before Y' ticket before. -thanks

File: client/app/components/flowchart/flowchart_directive.js (hunk 2 above)
Comment (AllenBW) [inline client/app/components/flowchart/flowchart_directive.js]: oh yeah totally! was saying the use of it in the directive appeared to also be easily swap-able with vm

File: client/app/states/designer/blueprints/editor/editor.html (hunk 1 above)
Comment (AllenBW) [inline client/app/states/designer/blueprints/editor/editor.html]: yep! totally wish em as directives/components, rid ourselves of that ng-controller in the html...   not sure its a straight forward task, hence the one of these days...

**Comment by dtaylor113** [comment]
Ok, do NOT do a 'merge X before Y comment' -got it

**Pull request merged**